### ISSUE TITLE
Overlay allows to click behind

### ISSUE THREAD

**Issue by veekeys** [iTwin/iTwinUI#826]
You can select the text behind overlay. Also if there was button, its possible to click it, which should not be allowed.

![Screenshot 2022-11-25 at 12 42 54](https://user-images.githubusercontent.com/81580355/203966043-1d787ebe-0322-4528-bb19-f4cbea92b432.png)

**Comment by mayank99** [comment]
Hey @veekeys, this overlay only exists in css currently. We could do some workarounds with `user-select` and `pointer-events` but I don't think that would be a simple change because we don't currently prescribe a class name for the wrapper (which also needs `position: relative`). It would also not prevent focusing any interactable elements behind the element.

I would suggest using the [`inert` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inert) on your content to fix this.

```html
<div style="position: relative">
   <div inert>...your content, button, etc...</div>
   <div class="iui-progress-indicator-overlay"></div>
</div>
```

Browser support for `inert` is pretty good but not perfect so I suggest using the [polyfill](https://github.com/WICG/inert) which is very reliable. The polyfill is only 8KB minified and you can load it conditionally to not affect bundle size for most users.
```js
<script type="module">
(async () => {
  // when not supported (firefox 107), this property will be undefined rather than false
  const isInertSupported = document.documentElement.inert === false;
  
  if (!isInertSupported) {
    await import('https://unpkg.com/browse/wicg-inert@3.1.2/dist/inert.min.js');
  }
})();
</script>
```

**Comment by veekeys** [comment]
Right, but what would be the downside of removing user-select now from overlay? It fixes issue of clicks, at least.
For focus, you would still consider full implementation with js (we do have FocusTrap available).

**Comment by mayank99** [comment]
> but what would be the downside of removing user-select now from overlay? It fixes issue of clicks, at least.

We don't have anywhere to add it to. Like I said, there is no wrapper class currently. The overlay is a sibling of the content.

> For focus, you would still consider full implementation with js (we do have FocusTrap available).

`FocusTrap` would not help, because it's the opposite of `inert`. We want to _prevent_ focus, not trap it.

**Comment by veekeys** [comment]
What?
This fixes the issue:
<img width="335" alt="Screenshot 2022-11-28 at 20 54 37" src="https://user-images.githubusercontent.com/81580355/204357810-9b8d92a5-a21c-42cb-9668-b316f228ed67.png">

We could use focus trap to force focus go into the overlay component and then never let it go out of it.

**Comment by mayank99** [comment]
> This fixes the issue: <img alt="Screenshot 2022-11-28 at 20 54 37" width="335" src="https://user-images.githubusercontent.com/81580355/204357810-9b8d92a5-a21c-42cb-9668-b316f228ed67.png">

You're right, I didn't see we had `pointer-events: none` on our overlay. Fixed in #827.

> We could use focus trap to force focus go into the overlay component and then never let it go out of it.

That is an accessibility anti-pattern as it would prevent the user from interacting with anything on the page while the overlay is shown. In general, keyboard handling should mimick mouse handling - if a user can interact with something using mouse, then they should be able to do it with keyboard too. Trapping focus within the overlay would break that rule. The correct solution is to use `inert`.
